crates/api_common/src/request.rs

```diff
@@ -206,6 +207,23 @@ pub async fn fetch_site_data(
       let missing_pictrs_file =
         |r: PictrsResponse| r.files.first().expect("missing pictrs file").file.clone();
 
+      let cache_remote_images = settings
+        .pictrs_config()
+        .map(|config| config.cache_remote_images)
+        .unwrap_or(true);
+      if !cache_remote_images {
+        return match is_image_content_type(client, url).await {
+          Ok(_) => {
+            let url = metadata_option
+              .clone()
+              .and_then(|metadata| metadata.image)
```

Comment:
These checks are already performed below in L232 . You might as well do the cache_remote_image check there, and do an early return there.

It would be nice if this whole section though could be refactored so that early returns aren't necessary. IE get rid of pictrs_hash and pictrs_thumbnail, and make it `thumbnail_url`, with exhaustive match conditions.

Reply:
Okay, I had a go at refactoring `fetch_site_data` to use pattern matching instead of early returns.